backend/routes.go

```diff
@@ -73,6 +75,41 @@ func (s *server) handleAddUser() http.HandlerFunc {
 	}
 }
 
+//GET: endpoint for all articles
+func (s *server) handleGetArticles() http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(w, "Requesting list of all articles")
+
+		type ArticleJson struct {
+			Title string `json:"title"`
+			URL   string `json:"url"`
+			//Votes []Vote `json:"votes"`
+		}
+
+		count, results, err := s.db.GetAllArticles()
+
+		if err != nil {
+			s.respond(w, req, makeErrorResponse(err), 500)
+			return
+		}
+
+		var articles = make([]ArticleJson, count)
+		var article ArticleJson
+		for i, a := range results {
+			article = ArticleJson{
+				Title: a.Title,
+				URL: a.URL,
+				//Votes: a.Votes
+			}
+			articles[i] = article
+		}
+		type ArticlesResponse struct {
```

Comment:
We can probably get rid of this type and pass the `articles` directly into the `respond` function! The respond function should work properly if given an array.